### ISSUE TITLE
DOC: 10 minutes to pandas Selection section `DataFrame.loc` followed by `()` 

### ISSUE THREAD

**Issue by ge-li** [pandas-dev/pandas#49254]
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/user_guide/10min.html

### Documentation problem

`DataFrame.loc`, etc should not be followed by parenthesis, because they are not functions. 

> # Selection
> While standard Python / NumPy expressions for selecting and setting are intuitive and come in handy for interactive work, for production code, we recommend the optimized pandas data access methods, [DataFrame.at()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.at.html#pandas.DataFrame.at), [DataFrame.iat()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iat.html#pandas.DataFrame.iat), [DataFrame.loc()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html#pandas.DataFrame.loc) and [DataFrame.iloc()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc).

> ## Selection by label
> See more in [Selection by Label](https://pandas.pydata.org/docs/user_guide/indexing.html#indexing-label) using [DataFrame.loc()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html#pandas.DataFrame.loc) or [DataFrame.at()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.at.html#pandas.DataFrame.at).

> ## Selection by position
> See more in [Selection by Position](https://pandas.pydata.org/docs/user_guide/indexing.html#indexing-integer) using [DataFrame.iloc()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc) or [DataFrame.at()](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.at.html#pandas.DataFrame.at).

### Suggested fix for documentation

> ... optimized pandas data access methods, [DataFrame.at](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.at.html#pandas.DataFrame.at), [DataFrame.iat](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iat.html#pandas.DataFrame.iat), [DataFrame.loc](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.loc.html#pandas.DataFrame.loc) and [DataFrame.iloc](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html#pandas.DataFrame.iloc).

**Comment by EleekaN** [comment]
Hi, I would like to work on these changes.

**Comment by mroeschke** [comment]
`loc` and `iloc` are technically also methods that can accept a method argument:

```
In [1]:         df = DataFrame([[6, "c", 10], [7, "d", 11], [8, "e", 12]])
   ...:

In [2]:         df.iloc(axis=0)[2] = 5
   ...:

In [3]: df
Out[3]:
   0  1   2
0  6  c  10
1  7  d  11
2  5  5   5

In [4]:         df = DataFrame([[6, "c", 10], [7, "d", 11], [8, "e", 12]])
   ...:

In [5]:         df.iloc(axis=1)[2] = 5
   ...:

In [6]: df
Out[6]:
   0  1  2
0  6  c  5
1  7  d  5
2  8  e  5
```

These example should be added in the `iloc` and `loc` docstrings instead of changing these references.

**Comment by mroeschke** [comment]
> Coding-style-wise, are df.iloc[2, :] and df.iloc[:, 2] preferred?

Yeah generally this is the preferred syntax for specifying which axis the label/integer should be applied

> Are there plans to unify loc and iloc such that the selection by labels or positions can be inferred?

Very unlikely. I think loc and iloc have existed for quite some time and have well defined boundaries for use cases (label based vs location based). Generally the library is trying to move toward more explicit behavior and avoid complex inference logic as it's harder to maintain & causes subtle bugs.
